lib/encoder.py

```diff
@@ -273,6 +284,19 @@ def ChangeValue(self, name, value):
     new_set.other_parts = self.other_parts
     return new_set
 
+  def RemoveValue(self, name):
+    """Return an OptionValueSet without the specified parameter."""
+    if not self.option_set.HasOption(name):
+      raise Error('Unknown option name %s' % name)
+    if self.option_set.Option(name).mandatory:
+      raise Error('Cannot remove option %s' % name)
+    new_set = OptionValueSet(self.option_set, "", self.formatter)
+    new_set.values = self.values.copy()
+    del new_set.values[name]
+    new_set.other_parts = self.other_parts
```

Comment:
This looks like a possible maintenance problem if OptionValueSet get more attributes in the future. Maybe you should move the copying logic to OptionValueSet and just do del new_set.values[name] after the copy?